pkg/greenboot/greenboot_test.go

```diff
@@ -0,0 +1,358 @@
+        "ostree.bootable" : true,
+        "rpmostree.inputhash" : "613038af0e67dfebb816b61b960bcb7e4f137050908e84f9ba726a4914aab319",
+        "rpmostree.rpmmd-repos" : []
+      },
+      "base-remote-replacements" : {      },
+      "layered-commit-meta" : {
+        "rpmostree.clientlayer" : true,
+        "version" : "38.20230414.n.0",
+        "rpmostree.clientlayer_version" : 6,
+        "rpmostree.state-sha512" : "a042a2bb9b453a335c4d30863f54d92d6509cd27ef687ecdd6b9b43121ee8c0aff1e341d445868e404b1089f4ae5273a284f0e5c10baef57e4fadc3efb9e5aa5",
+        "rpmostree.rpmmd-repos" : [],
+        "ostree.linux" : "6.2.9-300.fc38.x86_64",
+        "ostree.bootable" : true
+      },
+      "timestamp" : 1682329250,
+      "packages" : [ "bat", "greenboot", "langpacks-en", "neovim", "tmux", "vim" ],
+      "base-local-replacements" : [],
+      "osname" : "fedora",
+      "pending-base-checksum" : "233d5d86c58d4da70da4e1aec1c457c0b1b4a66fd5544d103d86a0280956e09d",
+      "pinned" : false,
+      "requested-modules-enabled" : [],
+      "modules" : [],
+      "booted" : false,
+      "base-removals" : [],
+      "unlocked" : "none",
+      "requested-base-removals" : [],
+      "base-checksum" : "7f13707d7180ddf167cf796fee7e1d3238fc20517a6cc3aa03108eb2d325a467",
+      "id" : "fedora-d1846d3f6fa89e6967cc740313ea727126e02db5b7af103f171d0e8f432a3d4d.0",
+      "origin" : "fedora:fedora/38/x86_64/silverblue",
+      "serial" : 0,
+      "base-timestamp" : 1681459461,
+      "gpg-enabled" : true,
+      "base-version" : "38.20230414.n.0",
+      "requested-local-packages" : [],
+      "checksum" : "d1846d3f6fa89e6967cc740313ea727126e02db5b7af103f171d0e8f432a3d4d",
+      "staged" : false
+    }
+  ],
+  "transaction" : null,
+  "cached-update" : null,
+  "update-driver" : null
+}`
+
+func Test_getOstreeStatus(t *testing.T) {
+	expectedStatus := ostreeStatus{
+		Deployments: []ostreeDeployment{
+			{
+				ID:     "fedora-8523da044806b21d28963a07c3e21e01d9a00dfeb76d4dbcdb582e8b3bf4d7b2.0",
+				Booted: false,
+			},
+			{
+				ID:     "fedora-551647113bff576a61185a056d7d2359751eac87375f2668955805e404cc6627.0",
+				Booted: true,
+			},
+			{
+				ID:     "fedora-d1846d3f6fa89e6967cc740313ea727126e02db5b7af103f171d0e8f432a3d4d.0",
+				Booted: false,
+			},
+		},
+	}
+
+	runOstreeStatus = func() ([]byte, error) {
+		return []byte(ostreeStatusJSON), nil
+	}
+
+	status, err := getOstreeStatus()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	if len(status.Deployments) != len(expectedStatus.Deployments) {
+		t.Fatalf("wrong number of deployments")
+	}
+
+	for _, d := range status.Deployments {
+		matching, found := func() (ostreeDeployment, bool) {
+			for _, ed := range expectedStatus.Deployments {
+				if d.ID == ed.ID {
+					return ed, true
+				}
+			}
+			return ostreeDeployment{}, false
+		}()
+
+		if !found {
+			t.Fatalf("deployment %s not found", d.ID)
+		}
+
+		if matching.Booted != d.Booted {
```

Comment:
Using the assert package for these tests will provide nice error messages when the tests fail without any special work on our part. See pkg/config/manifests_test.go for some examples.

Reply:
I don't see `pkg/config/manifests_test.go` file :sweat_smile: Switched to `assert`